index.js

```diff
@@ -1,18 +1,67 @@
 const readline = require('readline');
 const rl = readline.createInterface(process.stdin, process.stdout);
 
+// functions are here
+
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
 
+function binarySearch(min, max) {
+  return Math.floor((max + min) / 2);
+
+}
 start();
 
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+  console.log("Let's play a game where you (the human) think of a number between 1 and n and I (the computer) try to guess it.")
+
+  await ask("Press ENTER when you're ready. ");
+
+  // variables within the async function are here
+
+  let b = parseInt(await ask("Great. Please give me the upper range for my guesses. "));
+  
+  // I tried to set the default here, but it doesn't work
+  // if (b !== Number) {
+  //   b = 100;
+  // }
+
+  let a = 1;
+  let firstGuess = binarySearch(a, b);
+  let secondGuess = '';
+  let answerYesNo = "";
+  let answerHighLow = "";
+  let min = Math.min(a, b);
+  let max = Math.max(a, b);
+  let count = 1;
+
+  console.log("Now I'll guess a number between 1 and " + b + ".")
+  answerYesNo = await ask("Is it ... " + firstGuess + "? (Y/N) ")
+
+  // guessing loop here but it does not work correctly for all numbers and I don't know why
+
+  while (answerYesNo.toUpperCase() === "N" || answerYesNo.toUpperCase() === "NO") {
+    answerHighLow = await ask("Is it higher (H) or lower (L)? ")
+
+    if (answerHighLow.toUpperCase() === "H") {
```

Comment:
When you hit this statement, you know that the minimum number is now the computer's guess + 1. The minimum can't be what the computer just guessed, otherwise the user would have said yes, so the next lowest number is computer guess + 1. The weird behavior where the loop doesn't work for all numbers is probably coming from the next couple lines where you take the max value of two different values and then assign the 'a' variable whatever is in the secondGuess variable. The first time through, it will be an empty string, because that is what secondGuess is initialized as, so the second time through you are taking the max of an empty string and whatever firstGuess is. See me for more clarification if this is confusing, but this is where the issue is and where you can start your debugging.